azurelinuxagent/ga/monitor.py

```diff
@@ -445,7 +445,7 @@ def poll_telemetry_metrics(self):
 
             if metrics:
                 for metric in metrics:
-                    report_metric(metric.category, metric.counter, metric.instance, metric.value)
+                    report_metric(metric.category, metric.counter, metric.instance, float(metric.value))
```

Comment:
shouldn't we make the conversion when we initialize metric.value?

Reply:
The reasoning of casting the value here was to cast is at the time of writing the actual metric, as the count of sources adding metrics could change and it might miss to add the casting.

In `poll_all_tracked` or in other places in the future, where the metrics are populated, the number of metrics can change, which is why I added `float` casting here. I'll actually move it to the `report_metric` method itself, such that the data is always consistent and we don't hit schema mismatch issues later.

Eg: Didn't want to cast it at individual locations and do it at one place.
```
if cgroup.controller == "cpu":
   ...
    metrics.append(MetricValue("Process", "% Processor Time", cgroup.name, current_cpu_usage))
   ...
elif cgroup.controller == "memory":
   ...
   metrics.append(MetricValue("Memory", "Total Memory Usage", cgroup.name, current_memory_usage))
   ...
   metrics.append(MetricValue("Memory", "Max Memory Usage", cgroup.name, max_memory_usage))
   ...
```